package.json

```diff
@@ -1,21 +1,21 @@
 {
   "name": "matrix-js-sdk",
   "version": "2.4.6",
-  "description": "Matrix Client-Server SDK for Javascript",
-  "main": "index.js",
+  "description": "Matrix Client-Server SDK for JavaScript and TypeScript",
   "scripts": {
-    "test:watch": "jest spec/ --coverage --testEnvironment node --watch",
-    "test": "jest spec/ --coverage --testEnvironment node",
-    "gendoc": "jsdoc -c jsdoc.json -P package.json",
-    "start": "yarn start:init && yarn start:watch",
-    "start:watch": "babel src -s -w --skip-initial-build -d lib --verbose --extensions \".ts,.js\"",
-    "start:init": "babel src -s -d lib --verbose --extensions \".ts,.js\"",
+    "start": "echo THIS IS FOR LEGACY PURPOSES ONLY. && babel src -w -s -d lib --verbose --extensions \".ts,.js\"",
     "clean": "rimraf lib dist",
-    "build": "babel src -s -d lib --verbose --extensions \".ts,.js\" && rimraf dist && mkdir dist && browserify -d browser-index.js | exorcist dist/browser-matrix.js.map > dist/browser-matrix.js && terser -c -m -o dist/browser-matrix.min.js --source-map \"content='dist/browser-matrix.js.map'\" dist/browser-matrix.js",
-    "dist": "yarn build",
-    "watch": "watchify -d browser-index.js -o 'exorcist dist/browser-matrix.js.map > dist/browser-matrix.js' -v",
-    "lint": "eslint --max-warnings 93 src spec",
-    "prepare": "yarn clean && yarn build && git rev-parse HEAD > git-revision.txt"
+    "build": "yarn clean && git rev-parse HEAD > git-revision.txt && yarn build:compile && yarn build:compile-browser && yarn build:types",
+    "build:types": "tsc --emitDeclarationOnly",
+    "build:compile": "babel src -s -d lib --verbose --extensions \".ts,.js\"",
+    "build:compile-browser": "mkdirp dist && browserify -d browser-index.js -t [ babelify --sourceMaps=inline ] > dist/browser-matrix.js",
```

Comment:
They are not. Browserify picks up the source and does it all over again.

Reply:
huh - browser-index loads `lib/matrix` though, which would be the output of the `build:compile` step?

Reply:
It won't be shortly because that doesn't work.

Reply:
though making the change to have `browser-index` use `src` instead explodes on first sight, so this might not even be the right combination of transforms.

Reply:
@dbkr turns out that this has nothing to do with `lib` or `src`, it has to do with browserify being dumb.

Without babelify:
```
SyntaxError: 'import' and 'export' may appear only with 'sourceType: module' (34:0) while parsing F:\Workspaces\riot-web\matrix-js-sdk\browser-index.js while parsing file: F:\W
orkspaces\riot-web\matrix-js-sdk\browser-index.js
```
in other words: it can't support ES6 and babelify does something to it that makes it supported ES6 somehow?

The transform appears to be on load at least.

Reply:
Aha, I can well believe that browserify doesn't understand import - looks like it's more or less on life support. If we want to keep the bundled version we should probably swap it out for webpack to match the rest of our ecosystem, but not suggesting doing this now. So yeah, this is no worse than what we have now.

Reply:
Yea, webpack would be a good option. I'll look into it while I'm in the area, but won't make it a priority for sourcemaps being valid.